src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -36,13 +38,31 @@ public UUID send(LetterRequest letter, String serviceName) {
 
         final String messageId = generateChecksum(letter);
 
-        log.info("Generated message: id = {} for letter with print queue id = {}", messageId, letter.type);
+        log.info("Generated message: id = {}", messageId);
 
-        byte[] pdf = pdfCreator.create(letter);
-        Letter dbLetter = new Letter(messageId, serviceName, null, letter.type, pdf);
+        Optional<Letter> result = letterRepository.findByMessageIdOrderByCreatedAtDesc(messageId);
```

Comment:
Could you instead put a method on the repository to search by both MessageId and state ie:

if (repository.existsByMessageIdAndState(messageId, LetterState.Created)) {
}

Reply:
but i need to return the original letter in case there is a letter in `created` state. ought to be the latest hence the ordering by `created_at`

Reply:
You could do findByMessageIdAndStateOrderByCreatedAtDesc, then the filtering could be done by the DB.

Reply:
Could also use a spring projection to go straight to id as a UUID, but may not be worth it for this simple case - https://docs.spring.io/spring-data/rest/docs/current/reference/html/#projections-excerpts.projections